R/geom-violin.r

```diff
@@ -119,3 +136,22 @@ GeomViolin <- ggproto("GeomViolin", Geom,
 
   required_aes = c("x", "y")
 )
+
+# Returns a data.frame with info needed to draw quantile segments.
+create_quantile_segment_frame <- function(data, draw_quantiles) {
+  dens <- cumsum(data$density) / sum(data$density)
+  ecdf <- approxfun(dens, data$y)
+  ys <- ecdf(draw_quantiles) # these are all the y-values for quantiles
+
+  # Create data frame containing quantile segment data, one group per quantile
+  do.call ( rbind, lapply (ys, function(y_to_match) {
+    y_mismatches <- abs(y_to_match-data$y)
+    match_index <- which(y_mismatches==min(y_mismatches))[1]
```

Comment:
Wouldn't this be more easily done with `approxfun()`? The aesthetics within a violin are constant, so I don't think you need to do any matching up with rows in the original data